python/semantic_kernel/connectors/ai/function_choice_behavior.py

```diff
@@ -0,0 +1,244 @@
+
+    Class methods:
+        Auto: Returns FunctionChoiceAuto class with auto_invoke enabled, and the desired functions
+            based on either the specified filters or the full qualified names.
+        NoneInvoke: Returns FunctionChoiceNone class with auto_invoke disabled, and the desired functions
+            based on either the specified filters or the full qualified names.
+        Required: Returns FunctionChoiceRequired class with auto_invoke enabled, and the desired functions
+            based on either the specified filters or the full qualified names.
+    """
+
+    enable_kernel_functions: bool = True
+    maximum_auto_invoke_attempts: int = DEFAULT_MAX_AUTO_INVOKE_ATTEMPTS
+    filters: (
+        dict[Literal["excluded_plugins", "included_plugins", "excluded_functions", "included_functions"], list[str]]
+        | None
+    ) = None
+    function_fully_qualified_names: list[str] | None = None
+    type: FunctionChoiceType | None = None
+
+    @classmethod
+    def from_function_call_behavior(cls, behavior: "FunctionCallBehavior") -> "FunctionChoiceBehavior":
+        """Create a FunctionChoiceBehavior from a FunctionCallBehavior."""
+        from semantic_kernel.connectors.ai.function_call_behavior import (
+            EnabledFunctions,
+            KernelFunctions,
+            RequiredFunction,
+        )
+
+        if isinstance(behavior, (EnabledFunctions, KernelFunctions)):
+            return cls.Auto(
+                auto_invoke=behavior.auto_invoke_kernel_functions,
+                filters=behavior.filters if hasattr(behavior, "filters") else None,
+            )
+        if isinstance(behavior, (RequiredFunction)):
+            return cls.Required(
+                auto_invoke=behavior.auto_invoke_kernel_functions,
+                function_fully_qualified_names=[behavior.function_fully_qualified_name]
+                if hasattr(behavior, "function_fully_qualified_name")
+                else None,
+            )
+        return cls(
+            enable_kernel_functions=behavior.enable_kernel_functions,
+            maximum_auto_invoke_attempts=behavior.max_auto_invoke_attempts,
+        )
+
+    @property
+    def auto_invoke_kernel_functions(self):
+        """Return True if auto_invoke_kernel_functions is enabled."""
+        return self.maximum_auto_invoke_attempts > 0
+
+    @auto_invoke_kernel_functions.setter
+    def auto_invoke_kernel_functions(self, value: bool):
+        """Set the auto_invoke_kernel_functions property."""
+        self.maximum_auto_invoke_attempts = DEFAULT_MAX_AUTO_INVOKE_ATTEMPTS if value else 0
+
+    def _check_and_get_config(
+        self, kernel: "Kernel", fully_qualified_names: list[str] | None = None, filters: dict[str, Any] | None = {}
+    ) -> FunctionCallChoiceConfiguration:
+        """Check for missing functions and get the function call choice configuration."""
+        full_kernel_function_metadata = set(kernel.get_full_list_of_function_metadata())
```

Comment:
this can move to after `if fully_...` so that we don't have to do the work when not needed, or could we just pass the kernel to the _check_for_missing_function func?